### PR TITLE
Fix imported osu!stable scores being scaled by the classic scoring mode

### PR DESCRIPTION
**Pull request by smoogipoo**
Temporary fix for https://github.com/ppy/osu/issues/23782 / https://github.com/ppy/osu/issues/23756, preventing stable scores going into the billions when using the classic scoring mode.

It does NOT fix old lazer scores (which likely won't/can't be fixed), and does NOT convert SV1 to SV2 yet.

### CODE REVIEW

File: osu.Game/Database/RealmAccess.cs [hunk 1]
```diff
@@ -880,13 +884,41 @@ private void applyMigrationsForVersion(Migration migration, ulong targetVersion)
...
+                        using (var stream = files.Store.GetStream(replayFilename))
+                        {
+                            if (stream == null)
+                                continue;
+
+                            int version = new ReplayVersionParser().Parse(stream);
+                            if (version < LegacyScoreEncoder.FIRST_LAZER_VERSION)
+                                score.IsLegacyScore = true;
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (smoogipoo) [reply osu.Game/Database/RealmAccess.cs]: Added a try-catch.

File: osu.Game/Database/RealmAccess.cs [hunk 2]
```diff
@@ -880,13 +884,48 @@ private void applyMigrationsForVersion(Migration migration, ulong targetVersion)
...
+                    var scores = migration.NewRealm.All<ScoreInfo>();
+
+                    foreach (var score in scores)
+                    {
+                        string? replayFilename = score.Files.FirstOrDefault(f => f.Filename.EndsWith(@".osr", StringComparison.InvariantCultureIgnoreCase))?.File.GetStoragePath();
+
+                        if (replayFilename == null)
+                            continue;
+
+                        using (var stream = files.Store.GetStream(replayFilename))
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Database/RealmAccess.cs]: Have expanded the scope slightly. Apart from that, I think this is okay to go with. As I mentioned on discord, I don't think it's going to be a huge issue that this takes some time to run. It should be in the realm of seconds, and the startup time is already pretty long so I think users will be willing to wait.

[the remark this answers is not in the document]
Comment (smoogipoo) [reply osu.Game/Database/RealmAccess.cs]: Shouldn't that return the `null`? If not this function isn't properly documented.

**Review by peppy — DISMISSED** [review]

File: osu.Game/Database/RealmAccess.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Database/RealmAccess.cs]: Not sure what kind of documentation you're expecting, but the underlying `NativeStorage` doesn't do much apart from pass through stream functions from .NET:

https://github.com/ppy/osu-framework/blob/bd4ac6890a1ac7b2ec492b80975a0a2e61d9fcb0/osu.Framework/Platform/NativeStorage.cs#L88-L111

[the remark this answers is not in the document]
Comment (smoogipoo) [reply osu.Game/Database/RealmAccess.cs]: It needs to document that it both may return `null` if the path doesn't exist (also needs to be nullable annotated but that's a given), which is the case that @bdach was mentioning, and also use `<exception>` tags to describe that it also throws via the native calls.

**Review by bdach — APPROVED** [review]
Let's see how this pans out

**Pull request merged**